feat: add gap utility classes (#17715)
vuetifyjs/vuetify: packages/docs/src/pages/en/styles/spacing.md

### Horizontal

> Using the margin helper classes let you easily center content horizontally.

Reviewer: ```suggestion
Margin helper classes let you easily center content horizontally.
```

<example file="spacing/horizontal" />

### Gap

> Using the gap helper classes let you easily apply a gap between content.

Reviewer: ```suggestion
Use the gap helper classes to easily apply a gap between content.
```
